src/lib.rs

```diff
@@ -83,23 +97,29 @@ impl<'p, 'a> Serialize for SerializePyObject<'p, 'a> {
 
         macro_rules! add_to_map {
             ($map:ident, $key:ident, $value:ident) => {
-                if $key.is_none() {
-                    $map.serialize_key("null")?;
-                } else if let Ok(key) = $key.extract::<bool>() {
-                    $map.serialize_key(if key { "true" } else { "false" })?;
-                } else if let Ok(key) = $key.str() {
-                    let key = key.to_string();
-                    $map.serialize_key(&key)?;
+                if self.omit_none && $value.is_none() {
```

Comment:
for clarity, and to avoid too many changes, I would prefer this check was done inside the `$key.is_none()` if case, rather than as a separate block.

Reply:
I've tried to eliminate it but as I see there's no easy way?

Reply:
I changed it a bit https://github.com/samuelcolvin/rtoml/pull/31/commits/e7b8dddac60b9e5c02dc0504a3f2294c414631fe
But it still... not as plain as it'd better be :)